Modules/arraymodule.c

```diff
@@ -1130,7 +1130,7 @@ array_array_index(arrayobject *self, PyObject *v)
         cmp = PyObject_RichCompareBool(selfi, v, Py_EQ);
         Py_DECREF(selfi);
         if (cmp > 0) {
-            return PyLong_FromLong((long)i);
+            return PyLong_FromSsize_t(i); // BPO-41085: CPython was downcasting Py_ssize_t to long (Extra notes: On some platforms, long is the same size as int)
```

Comment:
The comment is useless, please remove it.

```suggestion
            return PyLong_FromSsize_t(i);
```